[BUG] git dependency installed with npm ci triggers error with npm prune --production

When installing some git dependencies using the Docker image node:10-alpine via `npm ci`, attempts to `npm prune --production` result in an error like:
```
npm ERR! code 128
npm ERR! Command failed: git clone --mirror -q https://github.com/nodecraft/ampify.git /root/.npm/_cacache/tmp/git-clone-6573274f/.git
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-6573274f/.git'
```

To help replicate the issue, I have a directory that contains a `package.json` and `package-lock.json`. I then have a `Dockerfile` which when building, illustrates the issue.

I've created a repository at https://github.com/Cherry/npm-prune-ci-bug detailing this, including commands to run to replicate the issue.

Last known working version: `6.10.3`. This issue does not occur if I use npm version `6.10.3` or earlier.
Tested versions that throw the above error:
```
6.11.0
6.11.1
6.11.2
6.11.3
6.12.1
6.13.1
```
It appears this regression was introduced in `6.11.0`, and exists through to the latest version.

If I can provide any further information, please let me know.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,13 @@
+## 6.13.6 (2020-01-09)
+
+### DEPENDENCIES
+
+* [`6dba897a1`](https://github.com/npm/cli/commit/6dba897a1e2d56388fb6df0c814b0bb85af366b4)
+  `pacote@9.5.12`:
+    * [`d2f4176`](https://github.com/npm/pacote/commit/d2f4176b6af393d7e29de27e9b638dbcbab9a0c7)
+      fix(git): Do not drop uid/gid when executing in root-owned directory
+      ([@isaacs](https://github.com/isaacs))
+
 ## 6.13.5 (2020-01-09)
 
 ### BUG FIXES

--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "version": "6.13.5",
+  "version": "6.13.6",
   "name": "npm",
   "description": "a package manager for JavaScript",
   "keywords": [

--- a/package.json
+++ b/package.json
@@ -110,7 +110,7 @@
     "once": "~1.4.0",
     "opener": "^1.5.1",
     "osenv": "^0.1.5",
-    "pacote": "^9.5.11",
+    "pacote": "^9.5.12",
     "path-is-inside": "~1.0.2",
     "promise-inflight": "~1.0.1",
     "qrcode-terminal": "^0.12.0",

